docs/pre-commit.md

```diff
@@ -0,0 +1,33 @@
+# Using pre-commit locally
+
+### Install and setup pre-commit
+- Install:
+  - Using pip: `pip install pre-commit`
+  - Using Homebrew: `brew install pre-commit`
+
+- Setup
+  - run `pre-commit install` to set up the git hook script
```

Comment:
I'd just link to the pre-commit docs, in case they change their instructions later

Reply:
Fully support both linking to docs and excerpting key text over here (in case the docs are moved or deleted later)